### ISSUE TITLE
Chinese path problem

### ISSUE THREAD

**Issue by chjq68** [pnpm/pnpm#6999]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

8.6.12

### Reproduction steps

Where user's windows accout include chinese char, yarn command could not execute because of UTF8 encode.

1. user name include chinse char, for example "中中中"
2. pnpm install -g yarn
3. yarn -v
4.  output is : 系统找不到指定的路径。(The system cannot find the path specified.)
5

### Describe the Bug

after inspect the yarn.CMD installed, I found the file encode is UTF-8. the file content  is correct but 
when execute under cmd, expected file encode is gbk.
After convert the yarm.cmd encode to gbk, command execute correctly.

### Expected Behavior

convert installed scritt such as yarn.cmd to gbk.
or add such command to the yarn.cmd : chcp 65001 (change page code to utf8)

### Which Node.js version are you using?

v18.17.1

### Which operating systems have you used?

- [ ] macOS
- [X] Windows
- [ ] Linux

**Comment by chjq68** [comment]
pnmp installed is standalone versoin.

**Comment by chjq68** [comment]
I try following way resolved the problem:
Before run pnpm setup, set env PNPM_HOME to other folder, then all work fines.
thank you.

**Comment by KSXGitHub** [comment]
I don't think this issue will ever be fixed without resorting to incredibly convoluted tricks in TypeScript code or rewriting almost all of pnpm in Rust (Rust handles path better than Node.js as it does not expect path being to be a string).

**Comment by zkochan** [comment]
Why? The suggested solution should be possible to do

>or add such command to the yarn.cmd : chcp 65001 (change page code to utf8)

it should be added to the generated command shim. The command shim is created by this package: https://github.com/pnpm/cmd-shim

**Comment by chjq68** [comment]
more information:
    npm.cmd no problem because it use "%~dp0" in npm.cmd. yarn.cmd not use "%~dp0" but real path, so problem occured.

**Comment by altaywtf** [comment]
I can't reproduce this on current pnpm (Windows 11, Node 24.21.0).

I ran `pnpm add -g` with the global dir under a path containing `中中ü`. On both 12.6.0 and 11.27.1 the generated `.cmd` shims in the global bin use `%~dp0`-relative paths and contain no non-ASCII bytes. They run under code pages 437, 936 and 65001.

Control: I rewrote the same shim with the absolute UTF-8 path, which is what the old shims did. Under 437 and 936 it fails with `MODULE_NOT_FOUND` and a mojibake path (`涓腑眉`). So the test does catch the original failure, and current shims avoid it.

Not tested: an actual Windows account whose user name has CJK characters (I used a CJK directory under an ASCII profile), and a real `yarn` global install (I used a small test package).

Unless someone can still reproduce it on 11.27.1 or 12.6.0, I think this can be closed.

Written by an agent (Claude Code, Opus 5.5).
